### ISSUE TITLE
Dynamic Implicit GEMM should not have Performance Configs or RunAndMeasure methods.

### ISSUE THREAD

**Issue by JehandadKhan** [ROCmSoftwarePlatform/MIOpen#332]
The Dynamic Implicit GEMM solvers are not tunable, but still have PerformanceConfig classes as well as RunAndMeasure Methods. 

This can result in unnecessary cycles spent in tuning.

**Comment by atamazov** [comment]
Wait, it looks like this is tunable. Its `GetSolution()` needs `PerformanceConfig` on input: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/38d18b7716a96d229373747eeda4c1a864a349e3/src/solver/conv_asm_implicit_gemm_v4r1_dynamic.cpp#L565-L575

**Comment by atamazov** [comment]
There are code quality issues (like unused boolean parameter of `GetSolution`, useless CO V3 related things, and ridiculous format of serialized `PerformanceConfig`), but I do not see why this is not tunable.

**Comment by carlushuang** [comment]
Hi @atamazov , ultimately dynamic kernel should not need tunable. When I implement this solver I just keep it compatible with the non-dynamic version of HIP igemm, and I think I need to re-factor this to avoid confusing.

**Comment by JehandadKhan** [comment]
Me and @carlushuang has a chat about this and he would factor this out in a future PR once he is done with other higher priority work.

**Comment by atamazov** [comment]
OMG indeed, `GetSolutionBase()` does not use `PerformanceConfig` to configure kernel. How can we write this :facepalm: ...

**Comment by carlushuang** [comment]
@atamazov sorry for the confusing, let me take the refactor into work.

**Comment by aserio** [comment]
@carlushuang Is this issue related to an open pull request?

**Comment by carlushuang** [comment]
> @carlushuang Is this issue related to an open pull request?

Hi, currently no PR is dealing with this issue. But there is another PR:https://github.com/ROCmSoftwarePlatform/MIOpen/pull/272 that changes several related invokers code. I prefer that PR merged then create another PR to deal with this issue, in case of code conflict :)

**Comment by aserio** [comment]
@carlushuang, As PR#272 is now merged have you returned to this issue?

**Comment by carlushuang** [comment]
@aserio Hi, yes let me take some time to fix this issue.

**Comment by carlushuang** [comment]
@aserio Please check https://github.com/ROCmSoftwarePlatform/MIOpen/pull/368

**Comment by aserio** [comment]
@carlushuang, Thanks for the update and your quick work!
